Scalar/CommandLine/UpgradeVerb.cs

```diff
@@ -142,13 +150,53 @@ private bool TryInitializeUpgrader(out string error)
             }
         }
 
+        private bool TryUpgradeWithBrew(out string error)
+        {
+            this.Output.WriteLine("Running 'brew update'");
+
+            var launcher = new ProcessLauncher();
+
+            bool result = launcher.TryStart("brew", "update", useShellExecute: false, out Exception ex);
+
+            if (!result) {
+                this.tracer.RelatedEvent(EventLevel.Warning, "Failure during 'brew update'", this.CreateEventMetadata(ex));
+            }
+
+            launcher.WaitForExit();
+
+            string packageName = "scalar";
+
+            if ((this.GetAvailableGitFeatures(this.tracer) & GitFeatureFlags.GvfsProtocol) != 0)
```

Comment:
If a user installs `scalar` and then later installs `microsoft-git`, they'll hit errors because Homebrew will try and upgrade the wrong cask (it won't be installed).

Similarly, if the user starts with `scalar-azrepos` and then gets a different Git installed on their path, then we'll try and upgrade `scalar`, and it'll fail.

It might be better to run `brew cask outdated` and scan for the presence of `scalar` or `scalar-azrepos` to see what is available to update. (This might be better for a background check to call `outdated`).

We could also just call `brew cask list` and look for `scalar` or `scalar-azrepos` to determine which flavour we are, and if we don't find either "this version was not installed by Homebrew" error message?

Reply:
Good plan!